src/ctxt.ml

```diff
@@ -3,37 +3,64 @@
 open Extra
 open Terms
 
-(** Representation of a typing context, associating a type (or [Term.term]) to
-    free [Bindlib] variables. *)
-type t = (tvar * term) list
+type t = ctxt
 
 (** [empty] is the empty context. *)
 let empty : t = []
 
-(** [add x a ctx] maps the variable [x] to the type [a] in [ctx]. *)
-let add : tvar -> term -> t -> t =
-  fun x a ctx -> (x,a)::ctx
+(** [assume h ctx] adds list of assumptions [h] mapping variables to their
+    types to context [ctx]. *)
+let assume : (tvar * term) list -> t -> t = fun hyps ctx ->
```

Comment:
why not replacing `(tvar * term) list` by `ctxt`?

Reply:
It was to have some kind of "interface", where the type of context is abstract. So even if the `ctxt` changes, use of `assume` does not.

Reply:
perhaps but it will be more uniform and simpler to use the type ctxt everywhere

Reply:
Ah my mistake, the ctxt types is a list of 3uples `tvar * term option * term`